resources/smpp/ra/src/main/java/org/restcomm/slee/resource/smpp/SmppSessionsImpl.java

```diff
@@ -77,79 +77,59 @@ public SmppTransaction sendRequestPdu(Esme esme, PduRequest request, long timeou
 			throw new NullPointerException("Underlying SmppSession is Null!");
 		}
 
+		SenderThread sender = esmeSenderThreads.get(esme.getName());
+		if (sender == null) {
+			throw new IllegalStateException("Esme sender not found");
```

Comment:
not sure reportinga RuntimeException is the best choice here. can we elaborate on the condition for this to happen? "a session was created,then we try to send req but no "sender" is foun in map" what may happen in the middle?is the key in map not present because of sbb logic or RA internal logic?

How is the existing invoking SBB logic affected?